### ISSUE TITLE
SegWit support

### ISSUE THREAD

**Issue by mikispag** [gcarq/rusty-blockparser#13]
Hi gcarq,

I would like to verify with you how compatible rusty-blockparser currently is with segwit blocks. It looks like required modifications should be minimal, if any.

If you need any help in making this happen, please let me know :) .

**Comment by gcarq** [comment]
Without digging into the code I think it only needs some minor adaptions to be compatible and of course the two new transaction types: 
* P2WPKH
* P2WSH

I'm linking the SegWit BIP for future reference https://github.com/bitcoin/bips/blob/master/bip-0141.mediawiki.
Currently I'm not very active on this project, so any help would be appreciated :)

**Comment by mikispag** [comment]
Sorry for the brevity, I am travelling, but I can confirm that the parser
is no longer working as of today.

I get:

```
[02:20:49] ERROR - dispatch: I/O Error: failed to fill whole buffer
```

**Comment by mikispag** [comment]
Hi gcarq,

any ETA for fixing the parser?
Do you need any help or debugging information to speed up the fix?

Thanks!

**Comment by uFire** [comment]
I confirm. 

rusty-blockparser unspentcsvdump /home/bitcoin/dump1/

always resulting to 

[00:41:38] ERROR - worker-0: I/O Error: failed to fill whole buffer
[00:41:39] ERROR - dispatch: I/O Error: failed to fill whole buffer

Tried on 2 machines with fresh Bitcoin Core 0.14.2

**Comment by mikispag** [comment]
FWIW, I have a very dirty block parsing "emergency [fix](https://github.com/mikispag/rusty-blockparser/commit/31c3111e20cdf5e6e209c7ade80eb110115110f9)" in my fork.

**Comment by gcarq** [comment]
Thanks a lot, I will implement this the next days

**Comment by mikispag** [comment]
Hey gcarq,

any update in implementing P2WPKH and P2WSH?
It would be awesome.

Thanks!

**Comment by gcarq** [comment]
SegWit support has been implemented in #62, it is fully functional for BTC, but not ready to merge yet.
